Shared/Samples/Set visibility of subtype sublayer/SetVisibilityOfSubtypeSublayerView.Model.swift

```diff
@@ -0,0 +1,164 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+extension SetVisibilityOfSubtypeSublayerView {
+    /// The view model for this sample.
+    class Model: ObservableObject {
+        /// A map with a streets basemap style.
+        let map = Map(basemapStyle: .arcGISStreetsNight)
+        
+        /// A feature table for the electrical network in this sample.
+        private let featureTable = ServiceFeatureTable(url: .featureServiceURL)
+        
+        /// A subtype feature layer created from the service feature table.
+        private let subtypeFeatureLayer: SubtypeFeatureLayer
+        
+        /// The subtype sublayer of the subtype feature layer in this sample.
+        private var subtypeSublayer: SubtypeSublayer?
+        
+        /// The renderer of the subtype feature layer.
+        private var originalRenderer: Renderer?
+        
+        /// The  subtype sublayer's label definition.
+        private(set) var labelDefinition: LabelDefinition = {
+            // Make and stylize the text symbol.
+            let textSymbol = TextSymbol(color: .blue, size: 10.5)
+            textSymbol.backgroundColor = .clear
+            textSymbol.outlineColor = .white
+            textSymbol.haloColor = .white
+            textSymbol.haloWidth = 2
+            // Make a label definition and adjust its properties.
+            let labelExpression = SimpleLabelExpression(simpleExpression: "[nominalvoltage]")
+            let labelDefinition = LabelDefinition(labelExpression: labelExpression, textSymbol: textSymbol)
+            labelDefinition.placement = .pointAboveRight
+            labelDefinition.usesCodedValues = true
+            return labelDefinition
+        }()
+        
+        // The formatter used to generate strings from scale values.
+        private let scaleFormatter: NumberFormatter = {
+            let numberFormatter = NumberFormatter()
+            numberFormatter.numberStyle = .decimal
+            numberFormatter.maximumFractionDigits = 0
+            return numberFormatter
+        }()
+        
+        /// A Boolean value indicating whether the settings should be presented.
+        @Published var isShowingSettings = false
+        
+        /// A Boolean value indicating whether to show the subtype sublayer.
+        @Published var showsSublayer = true
+        
+        /// A Boolean value indicating whether to show the subtype sublayer's renderer.
+        @Published var showsOriginalRenderer = true
+        
+        /// The current scale of the map.
+        var currentScale: Double = .zero
+        
+        /// The map's current scale value in text.
+        @Published var currentScaleText: String = ""
+        
+        /// The subtype sublayer's minimum scale value in text.
+        @Published private(set) var minimumScaleText: String = "Not Set"
```

Comment:
Instead of holding the scale texts, let's generate the text in the views.

The `currentScale` of the map view can be a state to the main view, and the `minScale` in the settings row can be computed from `subtypeSublayer?.minScale`.

Reply:
I did not apply this suggestion since the scale information is needed on the view and SettingsView, in which case the data is more easily accessible on the model.

Reply:
OK